FinderCompass/src/main/java/atomicstryker/findercompass/common/DefaultConfigFilePrinter.java

```diff
@@ -290,8 +292,14 @@ else if (settingList.isEmpty())
 			} else {
 				int[] ids;
 				for(int damage = 0; (ids=OreDictionary.getOreIDs(new ItemStack(pBlock, 1, damage))).length > 0; damage++){
+					boolean added = false;
 					for(int id:ids){
+						int size = oreDictIDs.size();
 						oreDictIDs.add(id);
+						added = added || size != oreDictIDs.size();

```

Comment:
This was the fastest way to stop the endless loop. My first code works with a few mods too, but we have a server with 250 mods and where was the endless loop.
It is not easy to debug this. The only reason for the endless loop were for different damages of same block same set of IDs. If all IDs are already in the list, nothing will be added and the size is the same (HashSet). More logical is the check with contains(), but size check should be (a little bit) faster.